lib/RedisRemoteSaiInterface.cpp

```diff
@@ -485,6 +493,114 @@ sai_status_t RedisRemoteSaiInterface::setRedisExtensionAttribute(
+        _In_ sai_object_id_t objectId,
+        _In_ const sai_redis_flex_counter_group_parameter_t *flexCounterGroupParam)
+{
+    SWSS_LOG_ENTER();
+
+    std::vector<swss::FieldValueTuple> entries;
+
+    if (flexCounterGroupParam == nullptr || !isSaiS8ListValidString(flexCounterGroupParam->counter_group_name, "COUNTER_GROUP_NAME"))
+    {
+        SWSS_LOG_ERROR("Invalid parameters when handling counter group operation");
+        return SAI_STATUS_FAILURE;
+    }
+
+    std::string key((const char*)flexCounterGroupParam->counter_group_name.list);
+
+    if (isSaiS8ListValidString(flexCounterGroupParam->poll_interval, POLL_INTERVAL_FIELD))
+    {
+        entries.emplace_back(POLL_INTERVAL_FIELD,
+                             std::string((const char*)flexCounterGroupParam->poll_interval.list, flexCounterGroupParam->poll_interval.count));
+    }
+
+    if (isSaiS8ListValidString(flexCounterGroupParam->stats_mode, STATS_MODE_FIELD))
+    {
+        entries.emplace_back(STATS_MODE_FIELD,
+                             std::string((const char*)flexCounterGroupParam->stats_mode.list, flexCounterGroupParam->stats_mode.count));
+    }
+
+    if (isSaiS8ListValidString(flexCounterGroupParam->plugins, "PLUGINS") && isSaiS8ListValidString(flexCounterGroupParam->plugin_name, "PLUGIN_NAME"))
+    {
+        entries.emplace_back(std::string((const char*)flexCounterGroupParam->plugin_name.list, flexCounterGroupParam->plugin_name.count),
+                             std::string((const char*)flexCounterGroupParam->plugins.list, flexCounterGroupParam->plugins.count));
+    }
+
+    if (isSaiS8ListValidString(flexCounterGroupParam->operation, FLEX_COUNTER_STATUS_FIELD))
+    {
+        entries.emplace_back(FLEX_COUNTER_STATUS_FIELD,
+                             std::string((const char*)flexCounterGroupParam->operation.list, flexCounterGroupParam->operation.count));
+    }
+
+    m_recorder->recordGenericSet(key, entries);
+
+    m_communicationChannel->set(key,
+                                entries,
+                                (entries.size() != 0) ? REDIS_FLEX_COUNTER_COMMAND_SET_GROUP : REDIS_FLEX_COUNTER_COMMAND_DEL_GROUP);
+
+    return waitForResponse(SAI_COMMON_API_SET);
+}
+
+sai_status_t RedisRemoteSaiInterface::notifyCounterOperations(
+        _In_ sai_object_id_t objectId,
+        _In_ const sai_redis_flex_counter_parameter_t *flexCounterParam)
+{
+    SWSS_LOG_ENTER();
+
+    if (flexCounterParam == nullptr || !isSaiS8ListValidString(flexCounterParam->counter_key, "COUNTER_KEY"))
+    {
+        SWSS_LOG_ERROR("Invalid parameters when handling counter operation");
+        return SAI_STATUS_FAILURE;
+    }
+
+    std::vector<swss::FieldValueTuple> entries;
+    std::string key((const char*)flexCounterParam->counter_key.list);
+    std::string command;
+
+    if (isSaiS8ListValidString(flexCounterParam->counter_ids, "COUNTER_IDS") && isSaiS8ListValidString(flexCounterParam->counter_field_name, "COUNTER_FIELD_NAME"))
```

Comment:
Please replace all COUNTERS IDS and other co stanowi string with proporcje defines or constexpr

Reply:
remove those const strings as they do not bring to many benefits. the const strings are for error messages but even without them a developer can easily find where goes wrong.